data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateGroup.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class AggregateGroup {
+    private final Map<Object, Object> groupState = new HashMap<>();
+
+    AggregateGroup() {
+
+    }
+
+    public Map<Object, Object> getGroupState() {
```

Comment:
I am worried we have created a class for the groupState but are still exposing the underlying data structure and therefore coupling the data to the processor and action.

Have we thought about providing methods / interface to alter the group state without exposing the underlying dats structure?

(I don't want to block this PR but I think this is good conversation to have given our community will be interacting with the state as they write aggregate actions.)

Reply:
It's a valid point. I have thought about it, and I just don't think we want to restrict the `AggregateAction` creators like that. It's going to be more work on their part, as instead of getting a java `Map` that they likely already understand and know how to interact with, they will get some Java class that we've made, that they will have to both read about to understand how to use, and may also find themselves limited in the control that they have over it.

Reply:
We definitely don't want to give them control of the `AggregateGroup` class, so the option is to extract the groupState map into a `GroupState` class that wraps the Map. However, this doesn't seem to add much value as we would just be adding functions for `put`, `get`, `putAll`, etc. that just call the actual Map functions, and there is nothing that I am aware of that we would really want to restrict the AggregateAction creator from calling on the Map. As far as validation goes, everything put into the Map will be turned into an Event, so the validation should happen there.

Reply:
This is a one way door decision that will couple the group state to the map interface. Maybe not completely one way door but it would require a breaking change to extend group state beyond the map interface. I just want to clarify, I do not want to limit the creator of an Aggregate Action from updating the group state as the desire. My biggest concern in raising this point is extensibility.

I don't want to debate the design here and I think others across the group may be interested in this conversation. I will raise an issue for this.